### PR TITLE
Migrate from binary db to MySQL

### PR DESCRIPTION
**Pull request by Headline**
Unfortunately we've made the horrible decision to use a binary file to store user credentials. While this was okay at the time, it crippled us at the knees in regards to future feature implementation. Since the serialized class cannot be edited, any additional features and their associated data would have to be stored in a new file, and a new serialized class. This is **horrible**, so we're migrating away now right before feature implementation begins. 

Cleverly, users who login will be added to our SQL server and then removed from the binary database file, successfully migrating that user from the badness that was that file. These changes don't touch the client, and are completely backwards-compatible with any client version. No interruptions should exist, and any that are experienced should be considered a bug.

🍻

### CODE REVIEW

File: server/src/com/michaelwflaherty/pulse/Client.java [hunk 1]
```diff
@@ -69,9 +72,16 @@ public void run()
                 if (params[0].equals("REGISTER")) // params[1] is usn, params[2] is salt, params[3] is hash
                 {
                     this.user = new UserInfo(params[1], params[3], params[2]);
```
Comment (Tuti) [inline server/src/com/michaelwflaherty/pulse/Client.java]: Although we can't switch the constructor, we should able to change the order of the params so we dont get 1, 3, 2

Comment (Headline) [reply server/src/com/michaelwflaherty/pulse/Client.java]: It looks a little funky, but we can't change parameter order without destroying client bcompat

File: server/src/com/michaelwflaherty/pulse/Client.java [hunk 2]
```diff
@@ -210,16 +243,49 @@ private void leaveRoom(String key)
...
     }
 
     public UserInfo findUser(String username)
     {
+        try
+        {
+            if (sql.userExists(username))
+            {
+                UserInfo info = sql.getUserInfo(username);
+                users.remove(info);
```
Comment (Tuti) [inline server/src/com/michaelwflaherty/pulse/Client.java]: You should leave a comment saying this is to remove the user info from the binary file

File: server/src/com/michaelwflaherty/pulse/Client.java (hunk 1 above)
Comment (Tuti) [reply server/src/com/michaelwflaherty/pulse/Client.java]: Then yea, something we'll just have to leave in place.

**Pull request merged**